api/src/app.controller.ts

```diff
@@ -1,12 +1,37 @@
-import { Controller, Get } from '@nestjs/common';
+import { Controller, Get, Post, Request, UseGuards } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
 import { AppService } from './app.service';
+import { Request as ExpressRequest } from 'express';
+import { LocalAuthGuard } from './auth/local-auth.guard';
+import { AuthService, AuthUser } from './auth/auth.service';
+import { JwtAuthGuard } from './auth/jwt-auth.guard';
 
-@Controller('hello')
+@Controller()
 export class AppController {
-  constructor(private readonly appService: AppService) {}
+  constructor(
+    private readonly appService: AppService,
+    private authService: AuthService,
+  ) {}
 
-  @Get()
+  @Get('hello')
   getHello(): string {
     return this.appService.getHello();
   }
+
+  @UseGuards(LocalAuthGuard)
+  @Post('auth/login')
```

Comment:
we should isolate controllers (`users`, `movies`, etc) once this gets too congested.